### ISSUE TITLE
operator status enum not rendered as string

### ISSUE THREAD

**Issue by ostromart** [istio/istio#21554]
There was a change a while back to remove the string representation of status because the enum should be rendered as string. However, I think this means logs, not status, which still shows int:
status:
  component_status:
    AddonComponents:
      status: 3

If that's the case we need to put the status string back (or just change the status to a string) because users will be looking at the CR status not the logs. Could you please verify @morvencao ?

**Comment by morvencao** [comment]
Double checked with latest code and kube version:
```
# kubectl version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-13T11:23:11Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2020-01-14T00:09:19Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
```

The status string can be rendered as string:
```
kubectl get iop
NAME                        AGE
example-istiocontrolplane   25s
# kubectl get iop example-istiocontrolplane -o yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"install.istio.io/v1alpha1","kind":"IstioOperator","metadata":{"annotations":{},"name":"example-istiocontrolplane","namespace":"default"},"spec":{"profile":"demo"}}
  creationTimestamp: "2020-02-27T06:57:04Z"
  finalizers:
  - istio-finalizer.install.istio.io
  generation: 1
  name: example-istiocontrolplane
  namespace: default
  resourceVersion: "1907"
  selfLink: /apis/install.istio.io/v1alpha1/namespaces/default/istiooperators/example-istiocontrolplane
  uid: 0d94e8dc-ea9e-42e1-bb2f-87f7aa06cd65
spec:
  profile: demo
status:
  componentStatus:
    AddonComponents:
      status: HEALTHY
    Base:
      status: HEALTHY
    EgressGateways:
      status: HEALTHY
    IngressGateways:
      status: HEALTHY
    Pilot:
      status: HEALTHY
  status: HEALTHY
```

**Comment by morvencao** [comment]
@ostromart mentioned this may be relevant with kubectl version, I will investigate that.

**Comment by morvencao** [comment]
@esnible I heard that you encounter the case that `status` can't rendered as string, can you provide some info about your env?

**Comment by morvencao** [comment]
@esnible I heard that you encounter the case that status can't rendered as string, can you provide some info about your env?

**Comment by morvencao** [comment]
@ostromart 

I finally reproduced this in `release-1.5` branch.
After checking all the changes of operator API, I think this is relevant to https://github.com/istio/api/pull/1263

I tried to build, deploy and test in-cluster operator with and without that PR, the results shows that the `status` fields need `UnmarshalJSON` and `MarshalJSON` function to be marshal/unmarshal normally.

Will cherrypick that PR to `release-1.5`.

**Comment by ostromart** [comment]
great! i'm still a bit puzzled why this was not showing up in your environment, but i'm glad you figured out the root cause.

**Comment by whitleykeith** [comment]
I'm still seeing this issue on 1.5.0. What do I need to do to resolve it?

**Comment by morvencao** [comment]
@whitleykeith I have verified the fix is already in branch `release-1.5`, so you have to wait for 1.5.1 release.
